File: stockservice/src/main/java/com/api/stockservice/domain/Entities/Product.java
package com.api.stockservice.domain.Entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.DynamicUpdate;
import org.hibernate.annotations.GenericGenerator;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@DynamicUpdate
public class Product {
    @Id
    @GeneratedValue(generator = "uuid")
    @GenericGenerator(name = "uuid", strategy = "uuid2")
    private String id;
    private String name;
    private String thumbnail;
    private String description;
    private Double price;
    private Integer quantity;

    @ManyToOne
    private Category category;

    @ManyToOne
    private Supplier supplier;
}

File: backend/services/MarketplaceService/src/MarketplaceService.Domain/Entities/Product.cs
using System.ComponentModel.DataAnnotations;

namespace MarketplaceService.Domain.Entities
{
    public class Product
    {
        [Key]
        public string ProductId { get; set; }
        public string MarqueName { get; set; }
        public string MarqueIcon { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public double Price { get; set; }
        public int Quantity { get; set; }
        public string Thumbnail { get; set; }
        public string CategoryId { get; set; }
        public Category? Category { get; set; }
        public List<CommandeProduct> CommandeProducts { get; set; } = new List<CommandeProduct>();
        public List<CartProduct> CartProducts { get; set; } = new List<CartProduct>();

    }
}
